### PR TITLE
housekeeping of Makefile

### PR DESCRIPTION
**Pull request by DGuhr**
* introduces `make apigen` command - creates code from proto files, creates v3 spec from v2 and validates the spec, in one command

* add describing comments to all commands

* adds go mod tidy to `make pr-check` command

* Fixes (and renames) the broken `make generate` command. Now called `make apiclient-gen` , which creates a go client from the spec in api/v1alpha/public/ - the  script pointed to a non-existent (outdated) spec file.

* introduces `make apiclient-delete` to delete the generated apiclient from `make apiclient-gen` in api/v1alpha/public

* aligns placement of PHONY to top of commands.

* Renames and aligns all api-related commands to start with either: apigen, apidocs, or apiclient

* Renames `make clean-tls` to `make tls-delete` to align with naming, and references this command in `make clean` instead of repeating the internals in make clean

### PR Template:

* [x] Are the agreed upon acceptance criteria fulfilled? (just housekeeping)

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [x] Do your changes have passing automated tests and sufficient observability?(does not apply, but checked every command

* [x] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**) (does not apply)

* [x] Are the agreed upon coding/architectural practices applied?

* [x] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production) (drive-by housekeeping, can create ticket if wanted)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?  (does not apply)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -62,45 +69,65 @@ tls-cert:
...
 
-generate:
+# delete the tls directory and certs
+.PHONY: tls-delete
+tls-delete:
+	@echo "removing generated tls certs"
+	@rm -rf tls/
+
+# generate a go client from the openAPI spec
+.PHONY: apiclient-gen
```
Comment (merlante) [inline Makefile:79]: I think this make target, apiclient-delete and the ./scripts/generate.sh can be deleted. I'm pretty sure we don't use it anywhere and is made redundant by buf-gen, etc. Might be good to take the opportunity to remove it.

File: Makefile [hunk 2]
```diff
@@ -132,42 +159,46 @@ endif
...
-clean:
-	@echo "removing all generated artifacts "
-	@rm -rf tls/
-	@cd api/gen/v1alpha && rm -rf .swagger-codegen/ && rm -f README.md && rm -f .swagger-codegen-ignore
-
-.PHONY: clean-apigenv3
-clean-apigenv3:
-	@echo "removing artifacts from openapi v3 generator"
-	@cd api/gen/v1alpha && rm -rf .swagger-codegen/ && rm -f README.md && rm -f .swagger-codegen-ignore
+clean: tls-delete apigen-v2-delete apiclient-delete
```
Comment (merlante) [inline Makefile:168]: Remove the binary too?

**Review by merlante — COMMENTED** [review]
Aside from comments, LGTM.

File: Makefile (hunk 1 above)
Comment (DGuhr) [reply Makefile:79]: ok, will do

File: Makefile (hunk 2 above)
Comment (DGuhr) [reply Makefile:168]: right, will do :)

**Comment by merlante** [comment]
Approved!

**Pull request merged**